extensions/amp-story/0.1/amp-story.js

```diff
@@ -427,28 +426,39 @@ export class AmpStory extends AMP.BaseElement {
 
   /** @private */
   assertAmpStoryExperiment_() {
-    if (!this.isAmpStoryEnabled_()) {
-      const errorIconEl = this.win.document.createElement('div');
-      errorIconEl.classList.add('i-amphtml-story-experiment-error-icon');
-
-      const errorMsgEl = this.win.document.createElement('span');
-      errorMsgEl.textContent = 'You must enable the amp-story experiment to ' +
-          'view this content.';
-
-      const experimentsLinkEl = this.win.document.createElement('a');
-      experimentsLinkEl.href = `${urls.cdn}/experiments.html`;
-      experimentsLinkEl.textContent = 'Enable or disable your experiments on ' +
-          'the experiments dashboard.';
-
-      const errorEl = this.win.document.createElement('div');
-      errorEl.classList.add('i-amphtml-story-experiment-error');
-      errorEl.appendChild(errorIconEl);
-      errorEl.appendChild(errorMsgEl);
-      errorEl.appendChild(experimentsLinkEl);
-      this.element.appendChild(errorEl);
-
-      user().error(TAG, 'enable amp-story experiment');
+    if (this.isAmpStoryEnabled_()) {
+      return;
     }
+
+    const errorIconEl = this.win.document.createElement('div');
+    errorIconEl.classList.add('i-amphtml-story-experiment-error-icon');
+
+    const errorMsgEl = this.win.document.createElement('span');
+    errorMsgEl.textContent = 'You must enable the amp-story experiment to ' +
+        'view this content.';
+
+    const experimentsLinkEl = this.win.document.createElement('button');
+    experimentsLinkEl.textContent = 'Enable';
+    experimentsLinkEl.addEventListener('click', () => {
+      this.enableAmpStoryExperiment_();
+    });
+
+    const errorEl = this.win.document.createElement('div');
+    errorEl.classList.add('i-amphtml-story-experiment-error');
+    errorEl.appendChild(errorIconEl);
+    errorEl.appendChild(errorMsgEl);
+    errorEl.appendChild(experimentsLinkEl);
+    this.element.appendChild(errorEl);
+
+    user().error(TAG, 'enable amp-story experiment');
+  }
+
+
+  /** @private */
+  enableAmpStoryExperiment_() {
+    toggleExperiment(this.win, 'amp-story', true);
+    toggleExperiment(this.win, 'dev-channel', true);
+    this.win.location.reload();
```

Comment:
Will this work on publisher origins? 

> If the assignment can't happen because of a security violation, a DOMException of type SECURITY_ERROR is thrown. This happens if the origin of the script calling the method is different from the origin of the page originally described by the Location object, usually when the script is hosted on a different domain.

https://developer.mozilla.org/en-US/docs/Web/API/Location/reload

Reply:
Replaced this with a screen that just instructs the user to refresh manually:

![screen shot 2017-10-31 at 3 32 50 pm](https://user-images.githubusercontent.com/1651960/32245091-087d4e4c-be51-11e7-81ca-5eb68733394e.png)